x-pack/plugin/data-frame/src/main/java/org/elasticsearch/xpack/dataframe/transforms/pivot/AggregationResultUtils.java

```diff
@@ -75,13 +75,15 @@
             aggNames.addAll(pipelineAggs.stream().map(PipelineAggregationBuilder::getName).collect(Collectors.toList()));
 
             for (String aggName: aggNames) {
-                final String fieldType = fieldTypeMap.get(aggName);
-
-                // TODO: support other aggregation types
                 Aggregation aggResult = bucket.getAggregations().get(aggName);
-
-                AggValueExtractor extractor = getExtractor(aggResult);
-                updateDocument(document, aggName, extractor.value(aggResult, fieldType));
+                // This indicates not that the value contained in the `aggResult` is null, but that the `aggResult` is not
+                // present at all in the `bucket.getAggregations`. This could occur in the case of a `bucket_selector` agg, which
+                // does not calculate a value, but instead manipulates other results.
+                if (aggResult != null) {
```

Comment:
The only valid case of this occurring is in `bucket_selector` aggregations. I cannot think of any code paths where the aggregation result will be `null` for when the async indexer is actually running. The only two cases I can think of are where
* The search request is some how mutated between being created by the PivotConfig and being executed
* The search response is corrupted, or mutated somehow before being processed
Both of these scenarios are undefined and I am not sure they are possible.